src/toil/job.py

```diff
@@ -1722,3 +1721,185 @@ def convertPromises(kwargs):
             elif isinstance(kwargs.get(r), PromisedRequirement):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
+
+
+def docker_call(job,
+                tool,
+                parameters=None,
+                work_dir='.',
+                docker_parameters=None,
+                outfile=None,
+                check_output=False,
+                defer=None):
+    """
+    Calls Docker for a particular tool with the specified parameters
+
+    Example of using docker_call in a Toil pipeline:
+        def toil_job(job):
+            work_dir = job.fileStore.getLocalTempDir()
+            path = job.fileStore.readGlobalFile(ref_id, os.path.join(work_dir, 'ref.fasta')
+            parameters = ['faidx', path]
+            docker_call(job, tool='quay.io/ucgc_cgl/samtools', work_dir=work_dir, parameters=parameters)
+
+    :param toil.Job.job job: The Job instance for the calling function.
+    :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools).
+    :param list[str] parameters: Command line arguments to be passed to the tool.
+    :param str work_dir: Directory to mount into the container via `-v`. Destination convention is /data
+    :param list[str,] docker_parameters: Parameters to pass to Docker. Default parameters are `--rm`,
+            `--log-driver none`, and the mountpoint `-v work_dir:/data` where /data is the destination convention.
+             These defaults are removed if docker_parmaters is passed, so be sure to pass them if they are desired.
+    :param file outfile: Pipe output of Docker call to file handle
+    :param bool check_output: When True, this function returns docker's output.
+    :param int defer: What action should be taken on the container upon job completion?
+           docker_call.FORGO (0) will leave the container untouched.
+           docker_call.STOP (1) will attempt to stop the container with `docker stop` (useful for
+           debugging).
+           docker_call.RM (2) will stop the container and then forcefully remove it from the system
+           using `docker rm -f`.
+           The default value is None and that shadows docker_call.FORGO, unless --rm is being passed in.
+    """
+    docker_call.FORGO = 0
+    docker_call.STOP = 1
+    docker_call.RM = 2
+
+    if parameters is None:
+        parameters = []
+
+    # Setup the outgoing subprocess call for docker
+    base_docker_call = ['docker', 'run']
+    if docker_parameters:
+        base_docker_call += docker_parameters
+    else:
+        base_docker_call += ['--rm', '--log-driver', 'none', '-v',
```

Comment:
We should document that these parameters are passed to docker in the docstring for this function.